### ISSUE TITLE
ERROR in ./~/react-html-email/lib/injectReactEmailAttributes.js Module not found: Error: Cannot resolve module 'react/lib/DOMProperty'

### ISSUE THREAD

**Issue by patrickpoh** [chromakode/react-html-email#30]
Hi, I experience this error after i upgrade the module. 

ERROR in ./~/react-html-email/lib/injectReactEmailAttributes.js
Module not found: Error: Cannot resolve module 'react/lib/DOMProperty'

this error shows up when im webpacking my project. Everything was ok before i updated it. I never change the code at all.

**Comment by mfeuermann** [comment]
I have same issue on different project. After update i have same issue. I also never change to code...
Im using "react-dom": "^15.0.0", "react-html-email": "^1.1.3", "react": "^15.0.0"

**Comment by patrickpoh** [comment]
@mfeuermann hi 5 man. However the code still works though

**Comment by mfeuermann** [comment]
Yes but im using Meteor.js and there is problem with building of application with this kind of problem. IM not able build Meteor app with this warning.

**Comment by Zaiban** [comment]
I'm having this too, also using Meteor.js.

**Comment by Luksys5** [comment]
Same here. Can't build project, although I don't use Meteor
using:
    react and react-dom 15.4.0
    babel-preset-react 6.16.0
    babel-core 6.18.2
    babel-loader-6.2.8
    webpack 1.13.3
    webpack-dev-server 1.16.2
    and other less important stuff...

**Comment by patrickpoh** [comment]
basically what is happening is because React 15.4.0 removed ReactInjection.js file. Hence this library broke

**Comment by ColeTownsend** [comment]
Thumbs up – looks like this used some private API stuff in react which now exists in `react-dom`. However, it appears to look for `react-dom` first and check for the api. 

```
require('react-dom');

// ensure base DOM properties are already injected

// FIXME: implement without relying on unstable import locations
var DOMProperty = undefined;
try {
  exports._DOMProperty = DOMProperty = require('react-dom/lib/DOMProperty');
} catch (e) {
  exports._DOMProperty = DOMProperty = require('react/lib/DOMProperty');
}
```

**Comment by Poky85** [comment]
In Webpack you can use this `webpack.config.js` as hotfix

```javascript
var config = {
	resolve: {
		alias: {
			"react/lib/DOMProperty": "react-dom/lib/DOMProperty"
		}
	}
};

export default config;
```

**Comment by wagnermoschini** [comment]
I had the same problem, setting an alias on `webpack.config.js` works fine!

**Comment by JuanchiRios** [comment]
I had the same problem. setting an alias on `webpack.config.js` like @wagnermoschini solve the problem

```
module.exports = {
    ...
    resolve: {
        alias: {
            "react/lib/DOMProperty": "react-dom/lib/DOMProperty"
        }
    }
};
```

**Comment by Batistleman** [comment]
Seems like this could be pull requested from: https://github.com/gabriprat/react-html-email

**Comment by intermundos** [comment]
Webpack alias works! Thank you.

**Comment by chromakode** [comment]
Hi all, I am working on a fix to this issue. The `require('react/lib/DOMProperty')` exists to support versions of React prior to 15.4.0. I did not anticipate this library would be built by a bundler such as webpack, which will attempt to resolve and bundle both modules. In the mean time, I think the webpack alias shared by @Poky85 should work.

**Comment by chromakode** [comment]
Just released v2.0.0 which drops support for older versions of React and should no longer have this issue.
